### PR TITLE
Feature/xcl 2521 stm32 f4 discovery board tutorial

### PR DESCRIPTION
**Pull request by atigyi**
[ Description ]
Initials for STM32F4 Xively Client build. Contains an eclipse project under examples/stm32/xively_demo_mt. Basic implementations for BSP TIME and RNG. LwIP BSD Socket API implementation for BSP IO NET. Cross compiling wolfSSL as well onto STM32 with an eclipse project.

[ JIRA ]
https://jira.3amlabs.net/browse/XCL-2660

[ Reviewer ]

[ QA ]
Travis CI target added for STM32 cross compilation on linux to produce libxively.a. Xively C Client manual execution on the board resulted in successful connection to Xively services.

[ Release Notes ]
Initial efforts made towards having a repo-built-in STM32 cross-compilation capability.

### CODE REVIEW

File: examples/stm32/WolfSSL/.settings/language.settings.xml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+			<provider copy-of="extension" id="org.eclipse.cdt.ui.UserLanguageSettingsProvider"/>
+			<provider-reference id="org.eclipse.cdt.core.ReferencedProjectsLanguageSettingsProvider" ref="shared-provider"/>
+			<provider-reference id="org.eclipse.cdt.managedbuilder.core.MBSLanguageSettingsProvider" ref="shared-provider"/>
+			<provider class="fr.ac6.mcu.ide.build.CrossBuiltinSpecsDetector" console="false" env-hash="1127902156830750966" id="fr.ac6.mcu.ide.build.CrossBuiltinSpecsDetector" keep-relative-paths="false" name="Ac6 SW4 STM32 MCU Built-in Compiler Settings" parameter="${COMMAND} ${FLAGS} -E -P -v -dD &quot;${INPUTS}&quot;" prefer-non-shared="true">
+				<language-scope id="org.eclipse.cdt.core.gcc"/>
+				<language-scope id="org.eclipse.cdt.core.g++"/>
+			</provider>
+		</extension>
+	</configuration>
+</project>
```
Comment (DellaBitta) [inline examples/stm32/WolfSSL/.settings/language.settings.xml]: could you make this xml less verbose?

File: examples/stm32/xively_demo_mt/Src/httpserver-socket.c [hunk 2]
```diff
@@ -0,0 +1,312 @@
+/**
```
Comment (DellaBitta) [inline examples/stm32/xively_demo_mt/Src/httpserver-socket.c]: Is this file really needed?  Do we need to include the http server socket?  It has a precompiled web page in here and everything.

Comment (atigyi) [inline examples/stm32/xively_demo_mt/Src/httpserver-socket.c]: good point. I've created a story for this:
https://jira.3amlabs.net/browse/XCL-2723
Reason I've created story for this is that the STM32 project is far away from its final state, and this issue is not the last one to solve. And this sprint is not primarily about STM32. Additionally I've created this PR only to save the state of the STM32 project into development branch.
Although this latter is not mandatory. So we can even leave STM32 state here on this branch without merging to development branch.
What do you think, does it worth to put this into development branch now?
